packages/deployment/app/cloudformation.template.yml

```diff
@@ -1018,12 +1018,13 @@ Resources:
             Value: {{{this}}}
       {{/if}}
     {{/each}}
-      {{# if ../ecs.docker}}
+    {{# ifEquals ../ecs.docker.registry "dockerhub"}}
```

Comment:
You make good points. These changes definitely would break the "default template behavior," which is something I'm wary of.

On the other hand, I'm not sure that failing a deployment that may create resources/dependencies that a user isn't expecting (if a user sets `registry` to `ecs`, for example) is a bad thing... I will agree that the absolute best option would be having kes fail during compilation instead of cloudformation sending some cryptic message.